src/main/java/at/ac/tuwien/kr/alpha/Main.java

```diff
@@ -81,10 +91,76 @@ public static void main(String[] args) {
 			Main.bailOut("Failed to parse program.", e);
 		}
 
-		Main.computeAndConsumeAnswerSets(alpha, ctx.getInputConfig(), program);
+		NormalProgram normalized = alpha.normalizeProgram(program);
+		InternalProgram preprocessed;
+		if (cfg.getInputConfig().isWriteDependencyGraph() || cfg.getInputConfig().isWriteComponentGraph()) {
+			LOGGER.debug("Performing program analysis in preparation for writing dependency and/or component graph file...");
+			AnalyzedProgram analyzed = AnalyzedProgram.analyzeNormalProgram(normalized);
```

Comment:
So the stratified evaluation is only conducted if writing dependency graph or component graph to an external file is enabled? I think this is a mistake.

Reply:
No, stratified evaluation is performed if and only if the flag ```evaluateStratifiedPart``` in ```Alpha```'s ```SystemConfig``` is set.

The purpose of this ```if``` and the overloaded versions of ```Alpha.performProgramPreprocessing``` is to only calculate dependency- and component graphs when they're needed. This is the case if either:
* dependency- or component graphs are to be written (but potentially no stratified evaluation performed)
* or stratified evaluation should be performed

In the first case, the normalized program is transformed into an ```AnalyzedProgram```, internally calculating dependency- and component graphs.
If no dependency- or component graphs should be written, ```alpha.performProgramPreprocessing``` is just called with the normalized program.
```performProgramPreprocessing``` then checks if stratified evaluation should be performed and, if so, triggers the dependency and component graph calculation beforehand.

Reply:
I changed it now as suggested.. TBH it still doesn't look much less complicated, but we're rid of `Alpha.performProgramPreprocessing(NormalProgram)` now.